lib/_stream_writable.js

```diff
@@ -831,12 +775,10 @@ const destroy = destroyImpl.destroy;
 Writable.prototype.destroy = function(err, cb) {
   const state = this._writableState;
   if (!state.destroyed) {
-    for (let entry = state.bufferedRequest; entry; entry = entry.next) {
-      process.nextTick(entry.callback, new ERR_STREAM_DESTROYED('write'));
+    for (const { callback } of state.getBuffer()) {
+      process.nextTick(callback, new ERR_STREAM_DESTROYED('write'));
     }
-    state.bufferedRequest = null;
-    state.lastBufferedRequest = null;
-    state.bufferedRequestCount = 0;
+    state.buffered = [];
```

Comment:
Nit: IIUC this also doesn't reset the index and `allBuffers` with `allNoop`, the stream is destroyed but IMO it'd be better to have the correct state anyway.